inception-search-mtas/src/main/java/de/tudarmstadt/ukp/inception/search/index/mtas/MtasUimaParser.java

```diff
@@ -232,78 +213,162 @@ else if (annotation instanceof Sentence) {
                                 
                                 // Add the UI annotation.feature name to the index as an annotation.
                                 // Replace spaces with underscore in the UI name.
-                                
                                 MtasToken mtasAnnotationFeature = new MtasTokenString(mtasId++,
-                                        annotationUiName.replace(" ", "_") + "."
-                                                + feature.getUiName().replace(" ", "_")
-                                                + MtasToken.DELIMITER + featureValue,
-                                        beginToken);
+                                        getIndexedName(annotationUiName) + "."
+                                        + getIndexedName(feature.getUiName()) 
+                                        + MtasToken.DELIMITER + featureValue, beginToken);
                                 mtasAnnotationFeature.setOffset(annotation.getBegin(),
                                         annotation.getEnd());
                                 mtasAnnotationFeature.addPositionRange(beginToken, endToken);
                                 tokenCollection.add(mtasAnnotationFeature);
+                                // Returns KB IRI label after checking if the  
+                                // feature type is associated with KB and feature value is not null
+                                String labelStr = "";
+                                if (feature.getType().contains(IndexingConstants.KB)
+                                        && (!featureValue.equals("null"))) {
+                                    labelStr = getUILabel(featureValue);
+                                }
+                                if (!labelStr.isEmpty()) {
+                                    String kbType = labelStr.split(MtasToken.DELIMITER)[0];
+                                    String kbValue = labelStr.split(MtasToken.DELIMITER)[1];
+                                    if (kbType.equals(IndexingConstants.kbConcept)) {
+                                        kbType = IndexingConstants.indexKBConcept;
+                                    }
+                                    else if (kbType.equals(IndexingConstants.kbInstance)) {
```

Comment:
Always compare `CONSTANT.equals(xxx)`, not `xxx.equals(CONSTANT)`. The first case is guaranteed to not create an NPE, the second not.

Reply:
DONE